internal/provider/data_source_organization_run_task.go

```diff
@@ -1,73 +1,142 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: MPL-2.0
 
-// NOTE: This is a legacy resource and should be migrated to the Plugin
-// Framework if substantial modifications are planned. See
-// docs/new-resources.md if planning to use this code as boilerplate for
-// a new resource.
-
 package provider
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"context"
+	"fmt"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func dataSourceTFEOrganizationRunTask() *schema.Resource {
-	return &schema.Resource{
-		Read: dataSourceTFEOrganizationRunTaskRead,
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &dataSourceOrganizationRunTask{}
+	_ datasource.DataSourceWithConfigure = &dataSourceOrganizationRunTask{}
+)
+
+// TODO: This model, and the following conversion function, need to be kept the same as the Org. Run Task Resource
+// (internal/provider/resource_tfe_organization_run_task.go) but it only differs by the HMAC Key. In the future we
+// should put in a change to add HMAC Key into this model and then we can share the struct. And more importantly we
+// we can use the later schema changes.
+type modelDataTFEOrganizationRunTaskV0 struct {
```

Comment:
We need a different model here to maintain the same schema as what was there before. Next PR will update this to use the v0 schema instead,